### PR TITLE
Added Dependency Detection to Fix Nix Flake builds

### PR DESCRIPTION
**Pull request by IllustratedMan-code**
Hi! I am new to Godot development. I am trying to create a nix flake, essentially a fancy build file that is exclusive to the NIX package manager. I had to add some dependency detection to the detect.py script to get it to work. As far as I know, the dependencies were already implied, so my changes should not break anything. I basically applied the patches I found [here](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/tools/godot). I would love it if someone could review my changes. Let me know if I need to do anything else!

### CODE REVIEW

File: platform/linuxbsd/detect.py [hunk 1]
```diff
@@ -340,6 +344,7 @@ def configure(env):
         if env["udev"]:
             if os.system("pkg-config --exists libudev") == 0:  # 0 means found
                 env.Append(CPPDEFINES=["UDEV_ENABLED"])
+                env.ParseConfig("pkg-config libudev --cflags --libs")
```
Comment (akien-mga) [inline platform/linuxbsd/detect.py:347]: This is also `dlopen`'ed, it's intentional that it's not linked.

**Comment by akien-mga** [comment]
These changes seem wrong, I would need to know more about the rationale. Godot builds fine on all Linux platforms without these so the problem seems specific to Nix and not to Godot's build configuration.

**Comment by IllustratedMan-code** [comment]
Interesting, this must be a nix specific problem then. I don't entirely understand why these particular additions work, but the nix builds seems to fail with missing header errors when they aren't present. I thought it was worth a shot anyway. It looks like I will have to resort to diff patches to get this to work. Thanks for the quick review! I'll go ahead and close this.

**Comment by akien-mga** [comment]
I'd be happy to help figure out what the actual issues are on Nix if you can provide with the error logs. Might be that we make assumptions on include paths that these patch fixes, and maybe the Nix headers are not in the standard location.

**Comment by akien-mga** [comment]
Could you maybe a bug report with details on the OS, build command used, and the error messages that you get? I have a hunch about what could cause an issue if Nix include paths are not the standard `/usr/include` stuff, but it will be best discussed in an issue.

**Comment by IllustratedMan-code** [comment]
I just posted a bug report [here](https://github.com/godotengine/godot/issues/59913). Thanks for all the help so far!

**Comment by aiswaryasankar** [comment]
**🔒 Entelligence AI Vulnerability Scanner**

✅ **No security vulnerabilities found!**

Your code passed our comprehensive security analysis.

📊 **Files Analyzed**: 1 files

---

**Pull request closed without merging**